app/views/layouts/application.html.erb

```diff
@@ -10,109 +10,112 @@
+    <div class="navbar-left footer">
+      <ul class="nav navbar-nav">
+        <!-- <li><a href="#"></a></li> -->
+        <% if user_signed_in? %>
+        <li id="user-dropdown-menu" class="dropdown">
+          <a id="user-dropdown-button" href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false"><%= "Hello #{current_user.first_name}" %><span class="caret"></span></a>
+          <ul id="user-dropdown-items" class="dropdown-menu bottom-up" role="menu">
+            <!-- <li><a href="#">Action</a></li> -->
+            <li><%= link_to "Edit Profile", edit_user_path(current_user) %></li>
+            <li><%= link_to "Edit Password", edit_password_change_path(current_user) %></li>
+            <!-- <li class="divider"></li> -->
+            <!-- <li><a href="#">Separated link</a></li> -->
+            <li class="divider"></li>
+            <!-- <li><a href="#">One more separated link</a></li> -->
+            <li><%= link_to "Log out", sessions_path, method: :delete %></li>
+          </ul>
+        </li>
+        <% else %>
+        <!-- <li><a href="#">Sign In</a></li> -->
+        <li><%= link_to "Sign In", new_session_path %></li>
+        <li> <%= link_to "Sign Up", new_user_path, id: "btn-sign-up" %></li>
+        <% end %>
+
+        <% if user_signed_in? %>
+          <% if current_user.missing_organization? %>
+            <li><%= link_to "Create Organization", new_organization_path %></li>
           <% else %>
-            <!-- <li><a href="#">Sign In</a></li> -->
-            <li><%= link_to "Sign In", new_session_path %></li>
-            <li> <%= link_to "Sign Up", new_user_path, id: "btn-sign-up" %></li>
+            <li><%= link_to "My Organization", organization_path(current_user.organization) %></li>
           <% end %>
-        </ul>
-        <ul class="nav navbar-nav">
-          <li><%= link_to "About", about_path	 %></li>
+        <% else %>
           <li><%= link_to "Create Organization", new_organization_path %></li>
-        </ul>
-      </div>
-      <div class="navbar-right footer-icons">
-        <i class="fa fa-twitter" aria-hidden="true"></i>
-        <i class="fa fa-facebook-official" aria-hidden="true"></i>
-        <i class="fa fa-google-plus" aria-hidden="true"></i>
-      </div>
-      <div class="navbar-right footer-text">
-          Built and maintained by students from <a href="http://codecore.ca">CodeCore</a>
-      </div>
+        <% end %>
+      </ul>
+      <ul class="nav navbar-nav">
+        <li><%= link_to "About", about_path	 %></li>
+      </ul>
+    </div>
+    <div class="navbar-right footer-icons">
+      <i class="fa fa-twitter" aria-hidden="true"></i>
+      <i class="fa fa-facebook-official" aria-hidden="true"></i>
+      <i class="fa fa-google-plus" aria-hidden="true"></i>
+    </div>
+    <div class="navbar-right footer-text">
+      Built and maintained by students from <a href="http://codecore.ca">CodeCore</a>
     </div>
```

Comment:
Hi Tam,

Lucas has completely reworked the nav-bar and it is only about 20 lines of code now. My team's push does not have his new version.